packages/strapi-design-system/src/TimePicker/TimePicker.js

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import TimeIcon from '@strapi/icons/Time';
+import styled from 'styled-components';
+import { Select, Option } from '../Select';
+
+const TimeIconWrapper = styled.div`
+  display: flex;
+  align-items: center;
+  svg {
+    height: 1rem;
+    width: 1rem;
+  }
+
+  svg path {
+    fill: ${({ theme }) => theme.colors.neutral500};
+  }
+`;
+
+export const TimePicker = ({
+  disabled,
+  error,
+  hint,
+  id,
+  onClear,
+  onChange,
+  value,
+  clearLabel,
+  label,
+  step,
+  ...props
+}) => {
+  const hoursCount = 24;
+  const times = [];
+  let min = 0;
+
+  for (let i = 0; i < hoursCount; i++) {
+    min = 0;
+
+    while (min < 60) {
+      times.push(`${i < 10 ? '0' + i : i}:${min < 10 ? '0' + min : min}`);
+      min += step;
+    }
+  }
+
+  return (
+    <Select
+      id={id}
+      label={label}
+      placeholder={'--:--'}
+      hint={hint}
+      onClear={onClear}
+      clearLabel={clearLabel}
+      error={error}
+      value={value}
+      onChange={onChange}
+      disabled={disabled}
+      startIcon={
+        <TimeIconWrapper>
+          <TimeIcon />
+        </TimeIconWrapper>
+      }
+      {...props}
```

Comment:
why do we need to explicitly pass the props `hint, label, error, value, disabled, etc` since it is implicitly passed in the props object?

Reply:
Exactly for what you mentioned: being explicit.

I like when it's easy to scan quickly which props is required in order for a component to work 😊 . Not having to jump between files and try to understand what is going under the hood.

I can spread the props object if you want, not against this, but this is the reason 😄